pkg/collector/corechecks/cluster/ksm/kubernetes_state_aggregators.go

```diff
@@ -126,6 +204,33 @@ func (a *counterAggregator) flush(sender aggregator.Sender, k *KSMCheck, labelJo
 	a.accumulator = make(map[[maxNumberOfAllowedLabels]string]float64)
 }
 
+func (a *lastCronJobCompleteAggregator) flush(sender aggregator.Sender, k *KSMCheck, labelJoiner *labelJoiner) {
+	a.aggregator.flush(sender, k, labelJoiner)
+}
+
+func (a *lastCronJobFailedAggregator) flush(sender aggregator.Sender, k *KSMCheck, labelJoiner *labelJoiner) {
+	a.aggregator.flush(sender, k, labelJoiner)
+}
+
+func (a *lastCronJobAggregator) flush(sender aggregator.Sender, k *KSMCheck, labelJoiner *labelJoiner) {
+	for cronjob, state := range a.accumulator {
+		hostname, tags := k.hostnameAndTags(
+			map[string]string{
+				"namespace":    cronjob.namespace,
+				"kube_cronjob": cronjob.name,
+			},
+			labelJoiner,
+			nil,
+		)
+
+		sender.ServiceCheck(ksmMetricPrefix+"cronjob.complete", state.state, hostname, tags, "")
+	}
+
+	a.accumulator = make(map[cronJob]cronJobState)
```

Comment:
does this mean we'll send a service check on every check run even?

Reply:
Would it be different from how the already existing `job.complete` service check is working: https://github.com/DataDog/datadog-agent/blob/3046093abdc80c5684d304d0018acf4dfa9ee838/pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go#L361 ?

Reply:
I just wanted to make sure I understood correctly. Thanks for confirming!